domain/model/modelmigration/import.go

```diff
@@ -247,21 +293,93 @@ func (i *importOperation) Execute(ctx context.Context, model description.Model)
 	return nil
 }
 
+// Execute will attempt to import the model constraints from description. If no
+// constraints have been set on the description then Execute will not attempt to
+// set any constraints for the model.
+func (i *importModelConstraintsOperation) Execute(
+	ctx context.Context,
+	model description.Model,
+) error {
+	cons := coreconstraints.Value{}
+	descCons := model.Constraints()
+
+	// It is possible that the constraints interface from description can be nil
+	// if it was never set. This isn't a documented contract of the description
+	// package but we include this check here for safety.
+	if descCons == nil {
+		return nil
+	}
+
+	if allocatePublicIP := descCons.AllocatePublicIP(); allocatePublicIP {
+		cons.AllocatePublicIP = &allocatePublicIP
+	}
+	if arch := descCons.Architecture(); arch != "" {
+		cons.Arch = &arch
+	}
+	if container := coreinstance.ContainerType(descCons.Container()); container != "" {
+		cons.Container = &container
+	}
+	if cores := descCons.CpuCores(); cores != 0 {
+		cons.CpuCores = &cores
+	}
+	if power := descCons.CpuPower(); power != 0 {
+		cons.CpuPower = &power
+	}
+	if inst := descCons.InstanceType(); inst != "" {
+		cons.InstanceType = &inst
+	}
+	if mem := descCons.Memory(); mem != 0 {
+		cons.Mem = &mem
+	}
+	if disk := descCons.RootDisk(); disk != 0 {
+		cons.RootDisk = &disk
+	}
+	if source := descCons.RootDiskSource(); source != "" {
+		cons.RootDiskSource = &source
+	}
+	if spaces := descCons.Spaces(); len(spaces) > 0 {
+		cons.Spaces = &spaces
```

Comment:
It seems we don't support negative spaces for model migration.

Reply:
We should. They will be encoded in the string suck as "^space1".